### PR TITLE
Document dynamic module splitting

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: doc/project/module.md [hunk 1]
```diff
@@ -170,33 +228,84 @@ There are currently two module split styles: `FewestModules` and `SmallestModule
...
 
 This also works for more than two public modules, creating intermediate shared (internal) modules as necessary.
 
+The dynamic import example above already assumes this module split style so a module listing is omitted.
+
 #### `SmallestModules`
 
 Create modules that are as small as possible.
-The smallest unit of splitting is a Scala class.
+The smallest unit of splitting is a Scala class (see [Splitting Granularity](#Splitting Granularity) below for more).
```
Comment (gzm0) [inline doc/project/module.md:250]: Is this the correct way to link to an anchor?

**Comment by gzm0** [comment]
I still need another self-review pass.

File: doc/project/module.md (hunk 1 above)
Comment (sjrd) [reply doc/project/module.md:250]: Yes, except that the titles are converted to something `#splitting-granularity`. I don't try to predict it, and instead generate then inspect the page with a browser's inspection tools...

File: doc/project/module.md [hunk 2]
```diff
@@ -151,15 +152,72 @@ Further, importing `b.js` will call `AppB.main`.
...
+In the example above, the `js.dynamicImport` is replaced by `import("./MyApp$$anon$1.js")`, followed by an invocation of the main entry point in `MyApp$$anon$1.js` (the `body` passed to `js.dynamicImport`).
+Therefore, when `main.js` is loaded, we do not need to load, nor download `MyApp$$anon$1.js`.
+It will only be loaded the first time `onClick` is actually called.
+This reduces the initial download time for users.
+
+Dynamic imports and entry points can be arbitrarily combined.
+
+### Module Split Styles
+
+So far, we have seen public modules and dynamic import boundaries can be defined.
```
Comment (sjrd) [inline doc/project/module.md:215]:
```suggestion
So far, we have seen how public modules and dynamic import boundaries can be defined.
```

File: doc/project/module.md [hunk 3]
```diff
@@ -170,33 +228,84 @@ There are currently two module split styles: `FewestModules` and `SmallestModule
...
 
 * `a.js`: public module, containing the export of `start`.
 * `b.js`: public module, containing the export of `start` and the call to `AppB.main`
 * many internal small modules (~50 for this example), approximately one per class.
 
+In the dynamic import example, this would generate:
+* `main.js`: public module, containing the export of `onClick`.
+* many internal small modules (~150 for this example), approximately one per class.
+
+Observe that in this split style, dynamic import boundaries do not affect splitting because the modules are already as small as possible.
```
Comment (sjrd) [inline doc/project/module.md:264]: I don't understand this sentence. :s

**Review by sjrd — CHANGES_REQUESTED** [review]

File: doc/project/module.md (hunk 3 above)
Comment (gzm0) [reply doc/project/module.md]: I have tried to clarify. PTAL

Comment (gzm0) [reply doc/project/module.md]: (maybe it doesn't need to be there and I can just drop it)

Comment (sjrd) [reply doc/project/module.md]: Hum, OK, now I understand. But I'm not sure it's clear for a user. Perhaps add something like "Dynamic import boundaries are still important for dynamic loading."? On a first reading, it sounds like this paragraph says that dynamic imports are irrelevant with SmallestModules. So it's confusing.

Perhaps even consider removing this paragraph altogether? I'm not sure it brings any relevant value to a user.

**Review by sjrd — APPROVED** [review]

**Pull request merged**